### DISCUSSION TITLE
Small question about DeepAtlas

### DISCUSSION THREAD

**Discussion by YuliangXiaoYLX** [Project-MONAI/tutorials#1601] [Q&A] [answered]
I am now working on the ```DeepAtlas``` tutorial and have a quick question about the ```warp``` and ```warp_nearest``` in the training. Since in the tutorial, you use the alternate training mode to save memory, which means one network will be frozen while the other one is active. In the segmentation network trail, the comment (shown below) said that we could use differentiable ```warp()``` instead of ```warp_nearest()``` to achieve the joint training, but this case, the ```registration network``` is frozen, how could we do the ```backpropagation``` and ```gradient descent``` for ```registration network``` based on this learnable ```warp()```? 

In my opinion, we should not freeze the ```registration network``` this time if we want to use ```warp()``` to train the registration jointly.

Hope someone could help me and thank you in advanced!

![Screenshot from 2024-01-02 20-45-59](https://github.com/Project-MONAI/tutorials/assets/70731556/dd52a419-a05a-459c-b02e-5e449d73e76b)

**Comment by KumoLiu** [comment]
Hi @ebrahimebrahim, could you please help share some comments here? I think it's this tutorial you've contributed for the first time.
Thanks in advance!

**Comment by ebrahimebrahim** [comment]
> In my opinion, we should not freeze the registration network this time if we want to use `warp()` to train the registration jointly.


Yes, that sounds right. We'd have to not freeze the registration network to do joint training.

**Reply by YuliangXiaoYLX** [reply]
@ebrahimebrahim Thanks for your response. Therefore, if we freeze the registration network, it does not matter which warp function we use, right? But warp_nearest won't involve gradient descent(not differentiable) calculations which decrease the computational burden.

**Reply by ebrahimebrahim** [reply, marked as answer]
That makes sense. But I don't know for sure if using `warp_nearest` decreases computational burden in this case. It might, but I would think that if you are keeping the registration network frozen then it doesn't matter if you use a differentiable warp function. Not certain about this.

**Reply by YuliangXiaoYLX** [reply]
Sounds good, thank you so much. I will have a try to test if that decreases the computational burden
